src/verifiers/sev-snp/crypto.c

```diff
@@ -0,0 +1,175 @@
+/* Copyright (c) 2021 Intel Corporation
+ * Copyright (c) 2020-2021 Alibaba Cloud
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <openssl/sha.h>
+#include <openssl/ts.h>
+#include <rats-tls/log.h>
+#include <string.h>
+#include "crypto.h"
+
+static bool digest_sha(const void *msg, size_t msg_len, uint8_t *digest, size_t digest_len,
+		       SHA_TYPE sha_type)
+{
+	if (sha_type == SHA_TYPE_256 && digest_len != SHA256_DIGEST_LENGTH)
+		return false;
+
+	if (sha_type == SHA_TYPE_256) {
+		SHA256_CTX context;
+
+		if (SHA256_Init(&context) != 1)
```

Comment:
rats-tls integrates crypto library which provides you with the necessary digest functions. So you should call crypto instance rather than calling openssl directly.

Reply:
what if there is no crypto instance, just using librats？